lib/session.js

```diff
@@ -108,6 +109,9 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 			if (user && (!passwordCheck || scmp(passwordCheck, hash(user.password)))) {
 				postHookedSigninWithUser(user, req, res, onSuccess, onFail);
 			} else {
+				if (!err) {
+					err = new Error('The password you entered is incorrect.');
```

Comment:
Looking at the code it doesn't really confirm one way or another if the user account exists. The return message could be a little more vague, I guess? Suggestions?

Additionally it appears the password is optional for this function anyways, so I'm really not sure anymore if there needs to be a more detailed error for the else side of this function. I'd be curious generally how secure this function is since it appears I can just pass in a user id and no password. It's entirely possible I'm missing something though...